[DOCPR-1022] Document manual workflow (#319)
canonical/sphinx-docs-starter-pack: docs/content/customise.rst

Customise the setup
===================

> Most configuration parameters can be set in the :file:`docs/conf.py` file.

Reviewer: ```suggestion
Configuration for starter pack based documentation is set in the :file:`docs/conf.py` file. This file is a pre-modified [Sphinx configuration file](https://www.sphinx-doc.org/en/master/usage/configuration.html) intended for use by all starter pack based documentation sets.
```

The starter pack's default configuration is prepared in a way that makes sense for most projects.

> However, you must set some critical parameters that may be unique for your project, like the project's name.

Reviewer: ```suggestion
However, you must set some critical parameters that are unique for your project, like the project's name.
```

In addition, you can find some optional parameters or add your own configuration parameters to the file.

Required customisation
----------------------

> You must check and update some of the parameters to the needs of your project.

Reviewer: ```suggestion
You must check and update some of the parameters specific to your project.
```

> Most of such mandatory parameters are commented with the `TODO` keyword.

Reviewer: ```suggestion
Mandatory parameters are commented with the `TODO` keyword.
```

If they're not we should find out soon enough :)

The following are some highlights of the available configuration parameters.

Update the project information
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Edit the :file:`docs/conf.py` file and update the configuration in the ``Project information`` section.
See the comments in the file for more information about each setting.

Open Graph configuration
^^^^^^^^^^^^^^^^^^^^^^^^

When you post a link to your documentation somewhere (for example, on Mattermost or Discourse), it might be shown with a preview.
This preview is configured through the Open Graph Protocol (:spellexception:`OGP`) configuration.

If you don't know yet where your documentation will be hosted, you can leave the URL empty.
If you do, specify the hosting URL.
You can leave the defaults for the website name and the preview image or specify your own.

Adjust the header
~~~~~~~~~~~~~~~~~

> The header is the top section of a page template.

Reviewer: I don't think we need to provide an explanation of what a header is.

Reviewer: Though contextually I don't think HTTP header would be a problem, H1/H2 are headings - not headers. Titling the section 'Adjust the page header' might work? But contextually the reader will understand it's not HTTP or H1 by reading the next sentence (contains product's tag, image, and name...).

> By default, the starter pack template header contains your product tag image, product name (taken from the ``project`` setting in the :file:`docs/conf.py` file), a link to your product page, and a drop-down menu for "More resources".

Reviewer: ```suggestion
By default, the starter pack template header contains your product's tag image and name (taken from the ``project`` setting in the :file:`docs/conf.py` file), a link to your product's page (if available), and a drop-down menu for "More resources".
```

> In many cases, this default setup is sufficient, but you can customise it.

Reviewer: ```suggestion
In many cases, this default setup is sufficient, but further customisation and configuration is available.
```
I'm not sure this is in the spirit of the original sentence here.

Deactivate the feedback button
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

By default, the starter pack includes a feedback button at the top of each page.
This button redirects users to your GitHub issues page, and populates an issue for them with details of the page they were on when they clicked the button.

If your project does not use GitHub issues, set the ``github_issues`` variable in the :file:`docs/conf.py` file to an empty value to disable both the feedback button and the issue link in the footer.

> If you want to deactivate only the feedback button, but keep the link in the footer, set ``disable_feedback_button`` in the :file:`docs/conf.py` file to ``True``.

Reviewer: ```suggestion
If you want to deactivate the feedback button, but keep the link in the footer, set ``disable_feedback_button`` in the :file:`docs/conf.py` file to ``True``.
```

If you rename a source file, its URL will change.
To prevent broken links, you should add a redirect from the old URL to the new URL in this case.

You can add redirects in the ``redirects`` variable in the :file:`docs/conf.py` file.

Configure included extensions
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The starter pack includes a set of extensions that are useful for all documentation sets.
They are pre-configured as needed, but you can customise their configuration in the  :file:`docs/conf.py` file.

The following extensions are included by default:

* ``canonical_sphinx``
* ``sphinxcontrib.cairosvgconverter``
* ``sphinx_last_updated_by_git``

The `canonical_sphinx` extension is required for the starter pack.

> It automatically enables the following extensions:

Reviewer: ```suggestion
It automatically enables and sets defaults for the following extensions if they are installed:
```

* ``custom-rst-roles``
* ``myst_parser``
* ``notfound.extension``
* ``related-links``
* ``sphinx_copybutton``
* ``sphinx_design``
* ``sphinx_reredirects``
* ``sphinx_tabs.tabs``
* ``sphinxcontrib.jquery``
* ``sphinxext.opengraph``
* ``terminal-output``
* ``youtube-links``

> To add new extensions needed for your documentation set, use the `extensions` parameter.

Reviewer: ```suggestion
To add new extensions needed for your documentation set, add them to the ``extensions`` parameter in :file:`docs/conf.py`.
```

> If the extensions need specific Python packages, add those to the :file:`.sphinx/requirements.txt` file.

Reviewer: ```suggestion
.. info::

    If any additional extensions need specific Python packages, ensure they are installed alongside the other requirements by adding them to the :file:`.sphinx/requirements.txt` file.
```

   [Page contents]

Possible values for the ``sequential_nav`` field are ``none``, ``prev``, ``next``, and ``both``.
See the :file:`docs/conf.py` file for more information.

Another example for page-specific configuration is the ``hide-toc`` field (provided by `Furo <Furo documentation_>`_), which can be used to hide the page-internal table of content.
See `Hiding Contents sidebar`_.

Add your own configuration
--------------------------

> A custom configuration parameter for your project can be used to extend or override the common configuration, or to define additional configuration that is not covered by the common ``conf.py`` file.

Reviewer: ```suggestion
Custom configuration parameters for your project can be used to extend or override the common configuration, or to define additional configuration that is not covered by the common ``conf.py`` file.
```
